pymatgen/io/pwmat/outputs.py

```diff
@@ -0,0 +1,304 @@
+from __future__ import annotations
+
+import copy
+import linecache
+from io import StringIO
+from typing import TYPE_CHECKING
+
+import numpy as np
+from monty.io import zopen
+from monty.json import MSONable
+
+from pymatgen.io.pwmat.inputs import ACstrExtractor, AtomConfig, LineLocator
+
+if TYPE_CHECKING:
+    from pymatgen.util.typing import PathLike
+
+
+__author__ = "Hanyu Liu"
+__email__ = "domainofbuaa@gmail.com"
+__date__ = "2023-12-28"
+
+
+class Movement(MSONable):
+    def __init__(self, filename: PathLike, ionic_step_skip: int | None = None, ionic_step_offset: int | None = None):
+        """
+        Description:
+            Extract information from MOVEMENT file which records trajectory during MD.
+        """
+        self.filename = filename
+        self.ionic_step_skip = ionic_step_skip
+        self.ionic_step_offset = ionic_step_offset
+        self.split_mark = "--------------------------------------"
+
+        self.chunk_sizes, self.chunk_starts = self._get_chunk_info()
+        self.n_ionic_steps = len(self.chunk_sizes)
+
+        self.ionic_steps: list[dict] = self._parse_sefv()
+        if self.ionic_step_offset and self.ionic_step_skip:
+            self.ionic_steps = self.ionic_steps[self.ionic_step_offset :: self.ionic_step_skip]
+
+    def _get_chunk_info(self):
+        """
+        Description:
+            Split MOVEMENT into many chunks, so that process it chunk by chunk.
+            Chunk contains the row of "--------------------------------------" .
+
+        Returns:
+            Tuple[List[int], List[int]]
+                chunksizes: List[int]
+                chunkstarts: List[int]
+        """
+        chunk_sizes: list[int] = []
+        row_idxs: list[int] = LineLocator.locate_all_lines(self.filename, self.split_mark)
+        chunk_sizes.append(row_idxs[0])
+        for ii in range(1, len(row_idxs)):
+            chunk_sizes.append(row_idxs[ii] - row_idxs[ii - 1])
+        chunk_sizes_bak: list[int] = copy.deepcopy(chunk_sizes)
+        chunk_sizes_bak.insert(0, 0)
+        chunk_starts: list[int] = np.cumsum(chunk_sizes_bak).tolist()
+        chunk_starts.pop(-1)
+        return chunk_sizes, chunk_starts
+
+    @property
+    def atom_configs(self):
+        """
+        Returns:
+            atom_configs: List[Structure]
```

Comment:
return type should come first, then description:

        Returns:
           list[Structure]: atom_configs